src/main/java/com/teksystems/TekcampExerciseSpring/Service/Implementation/UserServiceImplementation.java

```diff
@@ -0,0 +1,108 @@
+package com.teksystems.TekcampExerciseSpring.Service.Implementation;
+
+import java.util.Optional;
+
+import org.springframework.beans.BeanUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.stereotype.Service;
+
+import com.teksystems.TekcampExerciseSpring.Exception.ResourceNotFoundException;
+import com.teksystems.TekcampExerciseSpring.Models.UserEntity;
+import com.teksystems.TekcampExerciseSpring.Models.Response.UserResponse;
+import com.teksystems.TekcampExerciseSpring.Repositories.UserRepository;
+import com.teksystems.TekcampExerciseSpring.Services.UserService;
+import com.teksystems.TekcampExerciseSpring.Shared.Dto.UserDto;
+import com.teksystems.TekcampExerciseSpring.SharedUserId.SharedUserId;
+
+@Service
+public class UserServiceImplementation implements UserService{
+	private SharedUserId sharedUserId;
+	
+	public UserServiceImplementation(SharedUserId sharedUserId) {
+		this.sharedUserId = sharedUserId;
```

Comment:
strange name for a utility class...very ambiguous...I almost thought you were doing something dangerous